vyper/codegen/core.py

```diff
@@ -290,6 +290,65 @@ def append_dyn_array(darray_node, elem_node):
             return IRnode.from_list(b1.resolve(b2.resolve(ret)))
 
 
+def extend_dyn_array(context, dst_darray_node, src_darray_node):
+    assert isinstance(dst_darray_node.typ, DArrayType)
+    assert isinstance(src_darray_node.typ, DArrayType)
+
+    ret = ["seq"]
+    with dst_darray_node.cache_when_complex("dst_darray") as (
+        b1,
+        dst_darray_node,
+    ), src_darray_node.cache_when_complex("src_darray") as (b2, src_darray_node):
+        dst_len = get_dyn_array_count(dst_darray_node)
+        src_len = get_dyn_array_count(src_darray_node)
+        max_dst_len = dst_darray_node.typ.count
+
+        with dst_len.cache_when_complex("dst_darray_len") as (
+            b3,
+            dst_len,
+        ), src_len.cache_when_complex("src_darray_len") as (b4, src_len):
+            # Assert that dst darray is not at maximum length
+            dst_bounds_assertion = ["assert", ["lt", dst_len, max_dst_len]]
```

Comment:
this is not required. for instance, `xs.extend([])` may get rejected at runtime.